nemo_text_processing/text_normalization/zh/taggers/cardinal.py

```diff
@@ -0,0 +1,141 @@
+        3123    -> cardinal{ integer: "三千一百二十三"}
+        3,123   -> cardinal{ integer: "三千一百二十三"}
+        51234   -> cardinal{ integer: "五万一千二百三十四"}
+        51,234  -> cardinal{ integer: "五万一千二百三十四"}
+        0.125   -> cardinal{ integer: "零点一二五"}
+        self.fst:
+        123     -> cardinal{ integer: "一二三" }
+    '''
+
+    def __init__(self, deterministic: bool = True, lm: bool = False):
+        super().__init__(name="cardinal", kind="classify", deterministic=deterministic)
+        # base cardinal from digit to chinese
+        graph_digit = pynini.string_file(get_abs_path("data/number/digit.tsv"))
+        graph_zero = pynini.string_file(get_abs_path("data/number/zero.tsv"))
+        graph_teen = pynini.string_file(get_abs_path("data/number/digit_teen.tsv"))
+        graph_sign = pynini.string_file(get_abs_path("data/number/sign.tsv"))
+
+        graph_digit_with_zero = graph_digit | graph_zero  # from 0(read out) to 9
+        graph_no_zero = pynini.cross("0", "")  # cardinalzero,in some case we don't read it out
+        graph_digit_no_zero = graph_digit | graph_no_zero  # from 0(not read out) to 9
+        insert_zero = pynutil.insert('零')
+        delete_punct = pynutil.delete(",")  # to deal with 5,000,cardinalin english form
+
+        # 15 in 215
+        graph_ten_u = (graph_digit + pynutil.insert(UNIT_1e01) + graph_digit_no_zero) | (graph_zero + graph_digit)
+        # 15 only
+        graph_ten = (graph_teen + pynutil.insert(UNIT_1e01) + graph_digit_no_zero) | (graph_zero + graph_digit)
+        # 215 or 200
+        graph_hundred = (graph_digit + pynutil.insert(UNIT_1e02) + graph_ten_u) | (
+            graph_digit + pynutil.insert(UNIT_1e02) + graph_no_zero ** 2
+        )
+        # 3000 or 3002 or 3012 or 3123
+        graph_thousand = (
+            (graph_digit + pynutil.insert(UNIT_1e03) + graph_hundred)
+            | (
+                graph_digit
+                + pynutil.insert(UNIT_1e03)
+                + graph_zero
+                + graph_digit
+                + pynutil.insert(UNIT_1e01)
+                + graph_digit_no_zero
+            )
+            | (graph_digit + pynutil.insert(UNIT_1e03) + graph_zero + graph_no_zero + graph_digit)
+            | (graph_digit + pynutil.insert(UNIT_1e03) + graph_no_zero ** 3)
+        )
+        # just 3,000 or 3,002 or 3,012 or 3,123
+        graph_thousand_sign = (
+            (graph_digit + pynutil.insert(UNIT_1e03) + delete_punct + graph_hundred)
+            | (
+                graph_digit
+                + pynutil.insert(UNIT_1e03)
+                + delete_punct
+                + graph_zero
+                + graph_digit
+                + pynutil.insert(UNIT_1e01)
+                + graph_digit_no_zero
+            )
+            | (graph_digit + pynutil.insert(UNIT_1e03) + delete_punct + graph_zero + graph_no_zero + graph_digit)
+            | (graph_digit + pynutil.insert(UNIT_1e03) + delete_punct + graph_no_zero ** 3)
+        )
```

Comment:
```
|  Regexp:  This method:     Copy shortcuts:
|  /x?/     x.closure(0, 1)  x.ques
```

I think it's better to use `delete_punct.ques` to avoid the redundant code.